sam/bamWrite.go

```diff
@@ -307,16 +305,22 @@ func writeTriplet(bw *BamWriter, triplet []string) {
 
 	typ := triplet[1]
 	values := strings.Split(triplet[2], ",")
+	if len(values) == 1 && values[0] == "" {
+		values = nil
+	}
+	realTyp := typ
```

Comment:
Since realTyp is declared to be := typ, as opposed to a copy of typ, would changing realType to typ[1:2] on line 315 have weird effects since realTyp and typ might be linked? I'm just remembering that I've written code in the past where I had to make copies of similar values and avoid using "a := b".

Also, at code review we agreed that as soon as this bug fix gets merged into main, as long as we don't notice another bug, we will update to a new version of gonomics

Reply:
Sorry, it is less clear with the := operator, but `typ` is actually a `string` not a `[]string`. So `realTyp := typ` is copying `typ` not referencing it. So line 315 changes `realTyp` but not `typ`